test/integration/physics_system.cc

```diff
@@ -875,3 +875,110 @@ TEST_F(PhysicsSystemFixture, GetBoundingBox)
       ignition::math::Vector3d(-0.25, 2, 1)),
       bbox.begin()->second);
 }
+
+
+/////////////////////////////////////////////////
+// This tests whether nested models can be loaded correctly
+TEST_F(PhysicsSystemFixture, NestedModel)
```

Comment:
Would you mind adding a few more tests?

* loading a plugin for a nested model - bonus for loading it both from the SDF file and `ServerConfig` :slightly_smiling_face: 
* saving a world that has a nested model to make sure they're saved properly
* maybe also some test to make sure they can be logged and played back

Reply:
added more tests in a71a3d2e2e65bab10243987f91ed238a1329f7ab and 4a16b5585e9d3967229fd26da3bafeb803b4a86a